src/Agda2Hs/Compile/Type.hs

```diff
@@ -140,7 +140,11 @@ compileType t = do
       x  <- hsName <$> compileVar x
       return $ tApp (Hs.TyVar () x) vs
     Sort s -> return (Hs.TyStar ())
-    t -> genericDocError =<< text "Bad Haskell type:" <?> prettyTCM t
+    Lam argInfo restAbs
+      | usableModality argInfo  -> genericDocError =<< text "Bad Haskell type:" <?> prettyTCM t
+      | otherwise               -> compileType $ unAbs restAbs
```

Comment:
Use the `underAbstraction` utility function to make sure the TCEnv is updated properly (see the `Pi` case above).

Reply:
Here, I'm not sure we can do it: the first argument of `Lam` is only an `ArgInfo`, not a `Dom` as in `Pi`. I've tried to look for functions which create a `Dom Type` for that, but I couldn't find any. Or is `defaultArgDom` for that? But then, we would need a `Type`, which we still don't have for that part.

Reply:
You can use `underAbstraction_`, which creates a dummy domain (not perfect, but at least then the de Bruijn indices are not all messed up)

Reply:
That was what I was looking for:)